app/controllers/organizations/new.js

```diff
@@ -47,43 +51,161 @@ export default class OrganizationsNewController extends Controller {
 
   @action
   setNames(name) {
-    this.model.administrativeUnit.setAbbName(name);
+    this.currentOrganizationModel.setAbbName(name);
   }
 
   @action
   setAlternativeNames(names) {
-    this.model.administrativeUnit.setAlternativeName(names);
+    this.currentOrganizationModel.setAlternativeName(names);
   }
 
   @action
   setHasParticipants(units) {
-    this.model.administrativeUnit.hasParticipants = units;
+    this.currentOrganizationModel.hasParticipants = units;
   }
 
   @action
   setKbo(value) {
     this.model.structuredIdentifierKBO.localId = value;
   }
 
+  /**
+   * Update the {@link currentOrganizationModel} to the model type that matches
+   * the classification code selected by the user. This includes create a new
+   * model instance and instantiating it with the necessary values already
+   * filled in by the user. Any set relations with other organizations, for
+   * example the organization at hand was founded by another one, are not
+   * copied. These relations, and there interpretation, are typically specific
+   * to the classification of an organization, it is up to the user to set the
+   * necessary relations (again) via the form.
+   *
+   * @param {AdministrativeUnitClassificationCodeModel}
+   * organizationClassificationCode - the model object representing the
+   * classification code selected by the user
+   */
   @action
-  setClassification(value) {
-    this.model.administrativeUnit.classification = value;
-    this.model.administrativeUnit.subOrganizations = [];
-    this.model.administrativeUnit.foundedOrganizations = [];
-    this.model.administrativeUnit.isAssociatedWith = null;
-    this.model.administrativeUnit.isSubOrganizationOf = null;
-    this.model.administrativeUnit.wasFoundedByOrganizations = [];
-    this.model.administrativeUnit.hasParticipants = [];
+  organizationConverter(organizationClassificationCode) {
+    // Note: extra guard to avoid infinite loop when setting the recognised
+    // worship type ("Soort eredienst" field) triggers a refresh for the
+    // classification, which in turn triggers a refresh of recognized worship
+    // type and so on...
+    if (
+      organizationClassificationCode.id !==
+      this.currentOrganizationModel.classification.id
+    ) {
+      // Create new model instance based on the provided classification
+      let newOrganizationModelInstance = this.#createNewModelInstance(
+        organizationClassificationCode.id
+      );
+      // Note: explicitly set here to ensure form is updated
+      newOrganizationModelInstance.classification =
```

Comment:
**Nitpick:** I would move that line after the `copyPropertiesToModel` to avoid the classification to be overrided in `copyPropertiesToModel`

Reply:
That is indeed the more robust order of doing these things.